source/crud/tests/unified/bulkWrite-replaceOne-let.yml

```diff
@@ -0,0 +1,95 @@
+description: "BulkWrite replaceOne-let"
+
+schemaVersion: "1.0"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name crud-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+
+initialData: &initialData
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - { _id: 1 }
+      - { _id: 2 }
+
+tests:
+  - description: "BulkWrite replaceOne with let option"
+    runOnRequirements:
+      - minServerVersion: "5.0"
+    operations:
+      - object: *collection0
+        name: bulkWrite
+        arguments:
+          requests:
+            - replaceOne:
+                filter: &filter
+                  $expr:
+                    $eq: [ "$_id", "$$id" ]
+                replacement: &replacement {"x": 3}
+          let: &let
+            id: 1
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                update: *collection0Name
+                updates:
+                  - q: *filter
+                    u: *replacement
+                let: *let
+    outcome:
+      -
+        collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - { _id: 1, x: 3 }
+          - { _id: 2 }
+
+  - description: "Bulk Write replaceOne with let option unsupported
+  (server-side error)"
+    runOnRequirements:
+      - minServerVersion: "3.6.0"
+        maxServerVersion: "4.9"
+    operations:
+      - object: *collection0
+        name: bulkWrite
+        arguments:
+          requests:
+            - replaceOne:
+                filter: &filter1
+                  $expr:
+                    $eq: [ "$_id", "$$id" ]
+                replacement: &replacement {"_id": 3}
```

Comment:
If we keep this test, let's make sure it actually passes on pre-4.2 servers. In [DRIVERS-2210](https://jira.mongodb.org/browse/DRIVERS-2210) we realized that the server raises an error when using `let` without a pipeline-style update (i.e. `u` is an array). I'm not sure if an exception is made for replacement arguments.

Worst case we can just bump the `minServerVersion` on this test to 4.2.0.

 actually https://jira.mongodb.org/browse/DRIVERS-2183

Reply:
Bumped the `minServerVersion`.